packages/@aws-cdk/assertions/lib/private/section.ts

```diff
@@ -0,0 +1,56 @@
+import { Matcher, MatchResult } from '../matcher';
+
+export type MatchSuccess = { match: true, matches: any[] };
+export type MatchFailure = { match: false, closestResult?: MatchResult, analyzedCount: number };
+
+export function matchSection(section: any, matcher: Matcher): MatchSuccess | MatchFailure {
+  let closestResult: MatchResult | undefined = undefined;
+  let matching: any[] = [];
+  let count = 0;
+
+  eachEntryInSection(
+    section,
+
+    (entry) => {
+      const result = matcher.test(entry);
+      if (!result.hasFailed()) {
+        matching.push(entry);
+      } else {
+        count++;
+        if (closestResult === undefined || closestResult.failCount > result.failCount) {
+          closestResult = result;
+        }
+      }
+    },
+  );
+
+  if (matching.length > 0) {
+    return { match: true, matches: matching };
+  } else {
+    return { match: false, closestResult, analyzedCount: count };
+  }
+}
+
+function eachEntryInSection(
+  section: any,
+  cb: (entry: {[key: string]: any}) => void): void {
+
+  for (const logicalId of Object.keys(section ?? {})) {
+    const resource: { [key: string]: any } = section[logicalId];
+    cb(resource);
+  }
+}
+
+export function formatFailure(closestResult: MatchResult): string {
+  return [
+    'The closest result is:',
+    reindent(JSON.stringify(closestResult.target, undefined, 2)),
+    'with the following mismatches:',
+    ...closestResult.toHumanStrings().map(s => `\t${s}`),
+  ].join('\n');
+}
+
+function reindent(x: string, indent: number = 2): string {
```

Comment:
Maybe `leftpad` or `lpad` instead?

```suggestion
function leftpad(x: string, indent: number = 2): string {
```

Reply:
Good call.